src/CommunityToolkit.Maui.Core/Essentials/FileSaver/FileSaverResult.shared.cs

```diff
@@ -0,0 +1,18 @@
+using System.Diagnostics.CodeAnalysis;
+
+namespace CommunityToolkit.Maui.Storage;
+
+/// <summary>
+/// Result of the <see cref="IFileSaver"/>
+/// </summary>
+/// <param name="FilePath">Saved file path</param>
+/// <param name="Exception">Exception if operation failed</param>
+public record FileSaverResult(string? FilePath, Exception? Exception)
```

Comment:
What do you think about adding a method called `.EnsureSuccess()`, similar to [`HttpResponseMessage.EnsureSuccessStatusCode()`](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpresponsemessage.ensuresuccessstatuscode?view=net-7.0). 

It would throw an exception, like the previous "unsafe" method, and we can also add the `MemberNotNullWhen` attributes, like the `IsSuccessful` property.